pageserver/src/tenant/size.rs

```diff
@@ -364,12 +362,9 @@ async fn calculate_logical_size(
         .await
         .expect("global semaphore should not had been closed");
 
-    tokio::task::spawn_blocking(move || {
-        let _permit = permit;
-        let size_res = timeline.calculate_logical_size(lsn);
-        TimelineAtLsnSizeResult(timeline, lsn, size_res)
-    })
-    .await
+    let _permit = permit;
+    let size_res = timeline.calculate_logical_size(lsn).await;
+    Ok(TimelineAtLsnSizeResult(timeline, lsn, size_res))
```

Comment:
If `calculate_logical_size`  takes a long time, without having to download anything, this can now hog the async thread for a very long time. It will finish eventually though. I guess that's OK for now, it's a more general problem we have with our model where we freely perform std synchronous I/O in async code.

Reply:
I only changed this because `timeline.calculate_logical_size` **had** to become `async` because it now downloads layers internally.
Is there a way to tell Tokio that a future will be compute-heavy? I'd use it for this.

Reply:
Cool, will implement that.

Reply:
Ok, I thought this was straight-forward, and it is.
Then I looked at other callers of `Timeline::calculate_logical_size` and found that in `try_spawn_size_init_task` I removed the `spawn_blocking`  as well.
So, presumably, we have the same problem there.

So, will duplicate @koivunej 's snippet from above to there as well.

Reply:
Hm actually that's the only other caller, so, can move that into `Timeline::calculate_logical_size`.

Reply:
Oh no, can't share the code because `size.rs` has this semaphore :see_no_evil: